### ISSUE TITLE
Rewrite the surprisal calculator 

### ISSUE THREAD

**Issue by gfzhou** [vvoelz/voelzlab#21]
I think it's the time to rewrite the surprisal calculator. There are several ideas in my mind. The surprisal calculator should be a class object which contains many functions. Those functions should be able to at least accomplish the following tasks:
1. Calculate the surprisal value and the variance of one state.
2. Calculate the surprisal values and the variances of all the states
3. Estimate the populations of the states using different methods
4. Automatically pick out the top N states with the largest surprising values or the largest variances.
5. More other features......
Most of the functions already existed in the old scripts. We need to rewrite them in a better format and make the API easier to use.
Any thoughts?

**Comment by vvoelz** [comment]
Yes -- this sounds good.   Please take the lead on this!  If you get the ball rolling, I can rewrite some of my example scripts to use the new version.    Maybe we should put this new code it in its own repository?

**Comment by gfzhou** [comment]
Yes, I think I will take the lead and maybe I will put it in a new folder called surprisal3. I think we could put the new codes to its own repository after we rewrite them.

**Comment by vvoelz** [comment]
Great! Looking fwd to it
